packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -451,7 +451,7 @@ export class SocketModeClient extends EventEmitter {
     const options: WebSocket.ClientOptions = {
       perMessageDeflate: false,
       ...this.clientOptions.tls,
-      ...this.clientOptions.agent,
+      ...(this.clientOptions.agent !== undefined ? { agent: this.clientOptions.agent } : {}),
```

Comment:
Could be simplified to, if we are allowing explicit `undefined`:
```suggestion
      agent: this.clientOptions.agent
```